log4j-docgen/src/main/java/org/apache/logging/log4j/docgen/internal/DefaultSchemaGenerator.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.logging.log4j.docgen.internal;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.xml.XMLConstants;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+import org.apache.logging.log4j.docgen.AbstractType;
+import org.apache.logging.log4j.docgen.Description;
+import org.apache.logging.log4j.docgen.PluginAttribute;
+import org.apache.logging.log4j.docgen.PluginElement;
+import org.apache.logging.log4j.docgen.PluginSet;
+import org.apache.logging.log4j.docgen.PluginType;
+import org.apache.logging.log4j.docgen.ScalarType;
+import org.apache.logging.log4j.docgen.ScalarValue;
+import org.apache.logging.log4j.docgen.Type;
+import org.apache.logging.log4j.docgen.io.stax.PluginBundleStaxReader;
+import org.apache.logging.log4j.docgen.util.TypeLookup;
+import org.apache.logging.log4j.docgen.xsd.SchemaGenerator;
+import org.apache.logging.log4j.docgen.xsd.SchemaGeneratorRequest;
+
+@Singleton
+@Named("default")
+public class DefaultSchemaGenerator implements SchemaGenerator {
+
+    private static final String LOG4J_PREFIX = "log4j";
+    private static final String LOG4J_NAMESPACE = "http://logging.apache.org/log4j/2.0/config";
```

Comment:
Shouldn't this rather be `.../2.x/...`? (I am not sure either...)

Reply:
Depends on the release version. As the site layout is now, `2.x` is a symlink to the latest release. If we have multiple schema versions published, they should have different versions in the URL somewhere.

Reply:
Among the XML namespaces I know there are two schools:

- [Java EE](http://www.oracle.com/webfolder/technetwork/jsc/xml/ns/javaee/index.html), [Jakarta EE](https://jakarta.ee/xml/ns/jakartaee/) and [Spring](http://www.springframework.org/schema/) do not have a version number in the namespace; they have a `version` attribute on the main
- [Maven](https://maven.apache.org/xsd/) has a version number in the namespace, but the XSDs are at `https://maven.apache.org/xsd/`.

In every case, the version number is in the filename. I would use some neutral namespace like `http://logging.apache.org/xml/ns/config`.

Reply:
I liked the `http://logging.apache.org/xml/ns/config` idea. I would appreciate it if you can implement that change and resolve this conversation.